common/persistence/visibility/store/elasticsearch/query_interceptors.go

```diff
@@ -66,29 +74,50 @@ func NewValuesInterceptor(
 	namespaceName namespace.Name,
 	saTypeMap searchattribute.NameTypeMap,
 	searchAttributesMapperProvider searchattribute.MapperProvider,
+	nameInterceptor *nameInterceptor,
 ) *valuesInterceptor {
 	return &valuesInterceptor{
 		namespace:                      namespaceName,
 		searchAttributesTypeMap:        saTypeMap,
 		searchAttributesMapperProvider: searchAttributesMapperProvider,
+		nameInterceptor:                nameInterceptor,
 	}
 }
 
 func (ni *nameInterceptor) Name(name string, usage query.FieldNameUsage) (string, error) {
 	fieldName := name
-	if searchattribute.IsMappable(name) {
+
+	_, isCustom := ni.searchAttributesTypeMap.Custom()[fieldName]
+	if !isCustom && fieldName == searchattribute.ScheduleID {
```

Comment:
You didn't map yet to field name at this point, so isCustom is likely always false.

Reply:
I don't _think_ that's true, otherwise we would be seeing a lot of test failures...